### PR TITLE
Fix jsgen

### PR DESCRIPTION
**Pull request by juancarlospaco**
- Fix `jsgen` FrameInfo line numbers and filenames for code inside `templates`. Tiny diff. 
- Relevant for Stacktraces, debugging, etc.

#### Repro

```nim
import std/jsconsole
import somemodule

proc example() =
  console.log 1
  foo:
    console.log 3
  console.log 4

example()
```

```nim
# somemodule.nim
import std/jsconsole

template foo*(funcion) =
    console.log 2
    funcion
```

- `nim js example.nim` and read generated `example.js` then diff `example.js` of this fix vs `devel`.

### CODE REVIEW

**Comment by ringabout** [comment]
Well, does this fix https://github.com/nim-lang/Nim/issues/8367?

**Comment by juancarlospaco** [comment]
> Well, does this fix #8367?

In more than 1 place I seen problems with the line numbers inside templates, but yes.

File: compiler/jsgen.nim [hunk 1]
```diff
@@ -793,6 +793,7 @@ proc lineDir(config: ConfigRef, info: TLineInfo, line: int): Rope =
     rope(toFullPath(config, info)), rope(line), rope(info.toColumn)
   ]
 
+var previousFileName = newStringOfCap(20)  # For frameInfo inside templates.
```
Comment (Araq) [inline compiler/jsgen.nim]: make this a field of the `PProc`. And don't use `newStringOfCap`, the results is moved from `toFilename` into the variable anyway. Micro-optimizations require some understanding to apply them properly, you cannot just hack them into a codebase and hope the reviewer misses them. :P

**Comment by ringabout** [comment]
> but yes.

Please include "fixes #number" in the title and the commit next time so that the corresponding issue can be closed automatically.

**Comment by juancarlospaco** [comment]
Thanks this fixes stacktraces to be actually useful.

**Pull request merged**